astropy/io/fits/file.py

```diff
@@ -49,6 +50,9 @@
     'wb': 'ostream', 'wb+': 'update',
     'ab': 'ostream', 'ab+': 'append'}
 
+# A match indicates the file was opened in text mode, which is not allowed
+TEXT_RE = re.compile('^[rwa]((t?\+?)|(\+?t?))$')
```

Comment:
I've guessed that. Let's say that guide is not aimed at advanced contributors like you, but more for people who are new to git and the basic concepts of PR based development.